components/compliance-service/integration_test/reporting_server_list_suggestions_test.go

```diff
@@ -0,0 +1,206 @@
+			r.Profiles[0].Controls[0].Title = id
+			r.Profiles = r.Profiles[:1]
+			r.Profiles[0].Sha256 = id
+			r.Profiles[0].Title = id
+			r.Recipes = []string{id}
+			r.ReportUuid = id
+			r.Roles = []string{id}
+
+			reportIds[i] = id
+		})
+
+		require.NoError(t, err)
+	}
+
+	waitFor(func() bool {
+		response, _ := server.ListReports(everythingCtx, &reporting.Query{})
+
+		return response != nil && len(response.Reports) == n
+	})
+
+	reportsProjects := map[string][]string{
+		"project1": reportIds[1:3],
+		"project2": reportIds[2:5],
+		"project3": reportIds[3:],
+	}
+
+	projectRules := map[string]*iam_v2.ProjectRules{}
+	for k, v := range reportsProjects {
+		projectRules[k] = &iam_v2.ProjectRules{
+			Rules: []*iam_v2.ProjectRule{
+				{
+					Conditions: []*iam_v2.Condition{
+						{
+							Type:   rules_tags.RolesTag,
+							Values: v,
+						},
+					},
+				},
+			},
+		}
+	}
+
+	// Send a project rules update event
+	esJobID, err := suite.ingesticESClient.UpdateReportProjectsTags(everythingCtx, projectRules)
+	assert.Nil(t, err)
+
+	suite.WaitForESJobToComplete(esJobID)
+
+	suite.RefreshComplianceReportIndex()
+
+	esJobID, err = suite.ingesticESClient.UpdateSummaryProjectsTags(everythingCtx, projectRules)
+	assert.Nil(t, err)
+
+	suite.WaitForESJobToComplete(esJobID)
+
+	suite.RefreshComplianceSummaryIndex()
+
+	successCases := []struct {
+		description     string
+		allowedProjects []string
+		expectedIds     []string
+	}{
+		{
+			description:     "user has access to all projects",
+			allowedProjects: []string{authzConstants.AllProjectsExternalID},
+			expectedIds:     reportIds,
+		},
+		{
+			description:     "user has access to one project with reports",
+			allowedProjects: []string{"project1"},
+			expectedIds:     reportIds[1:3],
+		},
+		{
+			description:     "user has access to some projects with reports",
+			allowedProjects: []string{"project1", "project2"},
+			expectedIds:     reportIds[1:5],
+		},
+		{
+			description:     "user has access to projects without reports",
+			allowedProjects: []string{"project4", "project5"},
+			expectedIds:     []string{},
+		},
+		{
+			description:     "user has access to one project with reports and unassigned reports",
+			allowedProjects: []string{"project1", authzConstants.UnassignedProjectID},
+			expectedIds:     reportIds[:3],
+		},
+		{
+			description:     "user has access to some projects with reports and unassigned reports",
+			allowedProjects: []string{"project1", "project2", authzConstants.UnassignedProjectID},
+			expectedIds:     reportIds[:5],
+		},
+		{
+			description:     "user has access to projects without reports and unassigned reports",
+			allowedProjects: []string{"project4", "project5", authzConstants.UnassignedProjectID},
+			expectedIds:     reportIds[:1],
+		},
+		{
+			description:     "user has access to unassigned reports",
+			allowedProjects: []string{authzConstants.UnassignedProjectID},
+			expectedIds:     reportIds[:1],
+		},
+	}
```

Comment:
These were copied pretty much verbatim from https://github.com/chef/automate/blob/945ce3f47b126336ce3dadabe2992145ef14f696/components/compliance-service/integration_test/reporting_server_list_reports_test.go. Only the descriptions have been changed.